src/effects/effectchainslot.h

```diff
@@ -34,6 +34,9 @@ class EffectChainSlot : public QObject {
 
     EffectSlotPointer addEffectSlot(const QString& group);
     EffectSlotPointer getEffectSlot(unsigned int slotNumber);
+    const unsigned int numEffectSlots() const {
```

Comment:
This information shouldn't need to be exposed.

Reply:
I have used this in EffectsManager to refresh all the effect slots. We can remove this if we change the refreshing methods.